__e2e__/init.test.js

```diff
@@ -84,3 +84,38 @@ test('init --template file:/tmp/custom/template', () => {
 
   expect(stdout).toContain('Run instructions');
 });
+
+test('init --template with custom project path', () => {
+  const projectName = 'TestInit';
+  const customProjectPath = getTempDirectory('command-init-custom-path');
+
+  // Create directory
+  writeFiles(customProjectPath, {});
+
+  run(DIR, [
+    'init',
+    '--template',
+    'react-native-new-template',
+    projectName,
+    customProjectPath,
+  ]);
+
+  // make sure we don't leave garbage
+  expect(fs.readdirSync(customProjectPath)).toEqual([projectName]);
+  expect(fs.readdirSync(path.join(customProjectPath, projectName))).toEqual(
+    templateFiles,
+  );
+
+  const pkgJson = require(path.join(
+    customProjectPath,
+    projectName,
+    'package.json',
+  ));
+
+  expect(pkgJson).toMatchSnapshot(
+    'package.json contains necessary configuration',
+  );
+
+  // Remove custom project path
+  cleanup(customProjectPath);
```

Comment:
Please wrap this test in a `describe` and move this call to `afterEach` so it's called when the test fails